### PR TITLE
sleep: use UResult

### PR DESCRIPTION
**Pull request by sonng**
Part of a subtask for #2464

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
Nice, thanks! Could you also include line 73 in this change?

**Comment by sonng** [comment]
Thanks for looking at it, let me know if there's anything else I am missing :)

File: src/uu/sleep/src/sleep.rs [hunk 1]
```diff
@@ -61,15 +64,21 @@ pub fn uu_app() -> App<'static, 'static> {
...
+fn sleep(args: Vec<&str>) -> UResult<()> {
     let sleep_dur =
-        args.iter().fold(
+        args.iter().try_fold(
             Duration::new(0, 0),
             |result, arg| match uucore::parse_time::from_str(&arg[..]) {
-                Ok(m) => m + result,
-                Err(f) => crash!(1, "{}", f),
+                Ok(m) => Ok(m + result),
+                Err(f) => Err(f),
```
Comment (tertsdiepraam) [inline src/uu/sleep/src/sleep.rs:73]: Great work on this! I'm just wondering whether we can merge these `match` expressions. Maybe something like this:
```rust
let sleep_dur =
    args.iter().try_fold(
        Duration::new(0, 0),
        |result, arg| match uucore::parse_time::from_str(&arg[..]) {
            Ok(m) => Ok(m + result),
            Err(f) => Err(USimpleError::new(1, format("{}", err))),,
        }
    }?;

thread::sleep(sleep_dur);

**Review by tertsdiepraam — COMMENTED** [review]
Great start on removing that `crash!` call. I hadn't thought of using `try_fold` yet. Just one suggestion to make it a bit less verbose :)

File: src/uu/sleep/src/sleep.rs (hunk 1 above)
Comment (sonng) [reply src/uu/sleep/src/sleep.rs:73]: Ah interesting, so the `?` operator can be used on a `Result` to make it return earlier. Nice! 

So I ended up with 

```
fn sleep(args: Vec<&str>) -> UResult<()> {
    let sleep_dur =
        args.iter().try_fold(
            Duration::new(0, 0),
            |result, arg| match uucore::parse_time::from_str(&arg[..]) {
                Ok(m) => Ok(m + result),
                Err(f) => Err(USimpleError::new(1, format!("{}", f))),
            },
        )?;

    Ok(thread::sleep(sleep_dur))
}
```

However, I'm unsure what the convention is here on the return. Is it okay to wrap the `thread::sleep(sleep_dur)` with an `Ok` or does it have to be separate items .ie

```
thread::sleep(sleep_dur);
Ok(())
```

**Pull request merged**